ToolbarTest - HamburgerTest - AppointmentDetailsTest

ddeliastefania opened the pull request.

Commit:
New application commits

Commit:
First commit on ToolbalValidationTest.
I created 4 different packages with different classes and I belive that I  figured out how to use the Matchers class and how to check all the Views. But not at all.
The test is not working 100%. I have some issues, but I figured out how to fix them.
I used Constants class for all strings.
The code is not cleaned, you will find a lot of comments, but not spaces I hope.

Commit:
ToolbarValidationTest works.
I changed the name of methods, constants and matchers classes.
I cleaned the code.

turcumadalina commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Helpers/Constants.java:
```diff
@@ -0,0 +1,11 @@
+    public static final String MORE_OPTIONS = "More options";
+    public static final String ALPHANUMERIC_RANDOM_STRING = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+    public static final String COLLAPSE = "Collapse";
+    public static final String TRASH = "Trash";
```
You can find this text in strings.xml

Commit:
I changed some assertTrue/assertEquals messages.
I changed the trash matcher.
I changed some access modifiers.

mihaisava1986 commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Helpers/HelpersMethods.java:
```diff
@@ -0,0 +1,86 @@
+        onView(matcher).perform(click());
+    }
+
+    public static boolean isFirstMatcherToTheLeftOfSecondMatcher(Matcher<View> matcher1, Matcher<View> matcher2) {
```
Here you don't check if matcher1 is left of matcher2. You check if element identified with matcher1 is left of the element identified with matcher 2.

ddeliastefania replied:
Done.

mihaisava1986 commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Helpers/HelpersMethods.java:
```diff
@@ -0,0 +1,86 @@
+        }
+    }
+
+    public static boolean isFirstMatcherToTheRightOfSecondMatcher(Matcher<View> matcher1, Matcher<View> matcher2) {
```
Same as above.

ddeliastefania replied:
Done.

mihaisava1986 commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Screen/Home.java:
```diff
@@ -0,0 +1,43 @@
+        return HelpersMethods.isFirstMatcherToTheLeftOfSecondMatcher(withId(R.id.menu_sort), withContentDescription(Constants.MORE_OPTIONS));
+    }
+
+    public static void clickSearchButtonAndSearch() {
```
This method only clicks the search button. It doesn't search for items

ddeliastefania replied:
I changed.

mihaisava1986 commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Screen/Search.java:
```diff
@@ -0,0 +1,23 @@
+    public static boolean isMyNoteTextVisible() {
+        return HelpersMethods.isVisible(withId(R.id.note_title));
+    }
+
```
Empty line

ddeliastefania replied:
Done.

mihaisava1986 commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Screen/Home.java:
```diff
@@ -0,0 +1,43 @@
+
+
+
+
```
Many empty lines

ddeliastefania replied:
Done.

mihaisava1986 commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Screen/TextNote.java:
```diff
@@ -0,0 +1,62 @@
+        TextNote.clickContentBox();
+        onView(withId(R.id.detail_content)).perform(typeText(HelpersMethods.getRandomString(5)));
+        TextNote.clickReturnButton();
+
```
Empty line

ddeliastefania replied:
Done

Commit:
Removed empty spaces and modified some method names

alex26andutu commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Test/ToolbarValidationTest.java:
```diff
@@ -0,0 +1,57 @@
+        assertFalse("My note text is not displayed.", Home.isMyNoteTextVisible());
+    }
+}
+
```
Delete this line.

ddeliastefania replied:
Done.

turcumadalina commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Screen/TextNote.java:
```diff
@@ -0,0 +1,61 @@
+        HelpersMethods.clickAction(withText(R.string.trash));
+    }
+
+    public static void addRandomStringText() {
```
cc @mihaisava1986 @alex26andutu

ddeliastefania replied:
I changed -> addRandomText().

turcumadalina commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Screen/TextNote.java:
```diff
@@ -0,0 +1,61 @@
+        TextNote.clickReturnButton();
+    }
+
+    public static void addSpecificStringText() {
```
cc @mihaisava1986 @alex26andutu

ddeliastefania replied:
I changed -> addSpecificText().

turcumadalina commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Helpers/HelpersMethods.java:
```diff
@@ -0,0 +1,86 @@
+        final String AB = Constants.ALPHANUMERIC_RANDOM_STRING;
+
+        SecureRandom random = new SecureRandom();
+        StringBuilder sb = new StringBuilder(newItemStringLength);
```
"sb"?

ddeliastefania replied:
I changed -> stringBuilder.

Commit:
Renamed some method names and deleted empty spaces.

Commit:
Final test commit.
I changed typeNewItemAction() method and I cleaned the code. It's working.

Commit:
Added textBold check. It's working.

Commit:
Appointment Details Test almost done.
I created new screens and I have to finish only Step 6.

Commit:
Hamburger test ongoing.

Commit:
HamburgerTest done. It's working.
I created new screens, constants and I cleaned the code.
I modified some names of methods.

alex26andutu commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Screen/Home.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public static boolean isMyNoteTextVisible() {
+        return HelpersMethods.isVisible(allOf(withId(R.id.note_title), withText(Constants.MY_NOTE), isCompletelyDisplayed()));
```
You can use isCompletelyDisplayed() in HelperMethodes / isVsisible

ddeliastefania replied:
Ok.

Commit:
All tests.

Commit:
All tests.

alex26andutu commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Screen/Trash.java:
```diff
@@ -0,0 +1,28 @@
+        return HelpersMethods.isVisible(withText(Constants.PAY_THE_BILLS));
+    }
+}
+
```
Delete this line.

ddeliastefania replied:
Done.

alex26andutu commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Test/AppointmentDetailsTest.java:
```diff
@@ -0,0 +1,85 @@
+
+        // Step 12. Press back.
+        TextNote.clickReturnButton();
+        Thread.sleep(2000);
```
This is not a step. You can add a comment before this.

ddeliastefania replied:
Done

Commit:
2 lines removed and one comment for Thread.sleep added.

mihaisava1986 commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Helpers/HelpersMethods.java:
```diff
@@ -0,0 +1,140 @@
+    }
+
+    public static void typeNewItemAction(int xItemsAdded, int xPositionToAddNewText) {
+        for (int i = 0; i <= xItemsAdded; i++) {
```
You could use "i < xItemsAdded", because when you call this method you have to use an argument which the desired numbers to add -1. And this could be misleading.

ddeliastefania replied:
Done.

mihaisava1986 commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Helpers/HelpersMethods.java:
```diff
@@ -0,0 +1,140 @@
+                HelpersMethods.typeTextAction(withId(R.id.detail_title), Constants.MY_NOTE);
+                TextNote.clickReturnButton();
+            } else {
+
```
Empty line

ddeliastefania replied:
Removed.

mihaisava1986 reviewed (commented):
Why do you have a class with all tests?

turcumadalina commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Test/AllTests.java:
```diff
@@ -0,0 +1,201 @@
+
+public class AllTests extends StartTheApplication {
+    @Test
+    public void ToolbarTest() {
```
👎

turcumadalina commented on omniNotes/src/androidTest/java/it/feio/android/omninotes/Test/AllTests.java:
```diff
@@ -0,0 +1,201 @@
+    }
+
+    @Test
+    public void AppointmentDetailsValidationTest() throws InterruptedException {
```
👎

Commit:
Cleaned code after the review.

Commit:
All tests.
